### PR TITLE
[2378] Sort and Filter rounds

### LINKED ISSUE #2378: [Explorer Homepage] Explorer: Add sort/filtering functionality to Rounds tab
**User Story:**
As a donor 
I want to easily find rounds on Grants Stack 
So that I can easily find projects to support 

**Acceptance Criteria** 
GIVEN that I am on the "Rounds" tab 
WHEN I click on "Sort by" 
THEN I see sorting options that include the following: 

- Newest to Oldest (based on round creation date) 
- Oldest to Newest (based on round creation date) 
- Round End Date - Earliest to Latest (based on round end date) 
- Round End Date - Latest to Earliest (based on round end date) 
- Matching funds available - High to Low (based on matching funds available) 
- Matching funds available - Low to High (based on matching funds available) 

GIVEN that I am on the "Rounds" tab 
WHEN I click on "Filter" 
THEN I see filtering options that include the following: 

- Filter by chain (all supported chains) 
- Filter by grant round type (QF or Direct Grants) 
- Filter by status (taking applications, active, finished) 

GIVEN that I select two or more filters in the same category 
WHEN I look at the page 
THEN I see the toggle and section name says "Multiple" 
AND the results are a union of both filters 

GIVEN that I select two or more filters in different categories 
WHEN I look at the page 
THEN I see the toggle and section name says "Multiple" 
AND the results are an intersection of filters in different categories 

GIVEN that I want to remove a filter 
WHEN I click on the applied filter 
THEN it removes the filter 

**Product & Design Links:**
- [Figma](https://www.figma.com/file/1uxgU2jb0yyN11SUgxARmF/Explorer-S19?type=design&node-id=792-11219&mode=design&t=DVqmgnYkbAHatvGR-0)

**Tech Details:**

**Open Questions:**
* Are the sort by contributor and project options high impact? If not, should we scope out for now? 
* > A (h/t @melissa-neira): In early donation browsing research, users mentioned wanting a starting point in their browsing journey. a lot of users mentioned wanting to see which projects had the most contributors and which had the most funding in order to vet a project's legitimacy

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### LINKED ISSUE #2527: [Explorer Homepage] Small design tweaks
**Overview**
The new page looks amazing -- let's have Carl and Melissa pair on some small design tweaks to translate "this looks good in a figma file" to "this looks good on a web page".

I'm attaching a few screenshots of things I noticed that may not have translated perfectly, but please take this as directional vs exact feedback!

**Details**

Screenshot 2023-10-28 at 9 24 48 AM
Screenshot 2023-10-28 at 9 23 24 AM
Screenshot 2023-10-28 at 9 21 50 AM
Screenshot 2023-10-28 at 9 21 05 AM
Screenshot 2023-10-28 at 9 20 16 AM
Screenshot 2023-10-28 at 9 18 27 AM

### LINKED ISSUE #2528: [Explorer Redesign] UI and Functionality follow up ticket
### Tasks 
**General**
- [x] On the Explore Rounds tab, let's make sure we're only showing rounds that have been marked as public. 

**Explorer Homepage > Home tab > Donate Now section**
- [x] Sort "Donate Now" section by matching pool high to low (show the top 4 rounds with largest matching pools) 
> Rationale: On average, rounds with a low matching pool tend to be spam/test rounds 
- [x] When I click "View all" from the Donate Now section, it should take me to the "Explore Rounds" tab with the status="Active" and type="Quadratic Funding" selected by default 

**Explorer Homepage > Home tab > Rounds Ending Soon section**
- [x] Cap at one row (3 rounds) per Melissa's [designs](https://www.figma.com/file/1uxgU2jb0yyN11SUgxARmF/Explorer-S19?type=design&node-id=792-11219&mode=design&t=2WNEbfBl59Fgp8GQ-0)
- [x] Sort rounds by ending soonest to latest (currently, it's rounds that are ending latest) 
- [x] When I click "View all" from the Rounds Ending Soon section, it should take me to the Explore Rounds tab with sort "Round end earliest" applied 

**Explorer Homepage > Explore Rounds tab** 

Default state of Explore Rounds tab should be (if I just click on the Explore Rounds tab): 
- [ ] Only show Active rounds or rounds Taking Applications by default (the filters status="Active" and status="Taking Applications" should be applied) 
> Rationale: We don't want to load any past rounds (status="Finished") to keep it clean. The option is there for people who want to explore past rounds. 
- [x] Sort by matching pool - high to low

Other minor issues: 
- [x] When I select a filter, I should also be able to unselect it (currently, I can only select a filter and then I'm locked in) 
- [x] I should be able to select multiple filters in the same category, and have them compound (i.e. if I select PGN and Optimism, then I should see all PGN and Optimism rounds) 
- [x] I should be able to select multiple filters across different categories, and have them intersect (i.e. if I select QF and Taking Applications and PGN, then I should only see QF rounds on PGN that are taking applications still) 
- [x] The first time I select a filter, I'm seeing an extra "red check mark" 

Screen Shot 2023-10-31 at 4 24 58 PM

### PR DESCRIPTION
**Pull request by carlbarrdahl**
- https://github.com/gitcoinco/grants-stack/issues/2378

## PR checklist

For every PR, make sure that these statements are true:
- [ ] Includes only changes relevant to the original ticket. Significant refactoring needs to be separated.
- [ ] Doesn't contain type casts and non-null assertions.
- [ ] Doesn't add `@ts-ignore`.
- [ ] Doesn't disable lints.
- [ ] Doesn't use `useState` just for computation - use plain variables instead.
- [ ] Splits components into pure components that don't depend on external state or hooks.
- [ ] Avoid embedding components within other components
- [ ] Doesn't propagate optional values without good reason, doesn't mark property values as optional if that doesn't represent reality.
- [ ] Doesn't duplicate existing code.
- [ ] Parses out-of-domain data - this includes user input, API respones, on-chain data etc.
- [ ] Doesn't contain commented out code.
- [ ] Doesn't contain skipped or empty tests.
- [ ] If this PR adds/updates any feature, it adds/updates its test script

Subjective - at the discretion of the reviewers
- Does things as simply as possible, but not simpler.
- Doesn't reinvent the wheel or create premature abstractions.

##### Refers/Fixes

- fixes #2378
- fixes #2527
- fixes #2528

### CODE REVIEW

File: packages/grant-explorer/src/features/api/rounds.ts [hunk 1]
```diff
@@ -155,24 +158,27 @@ export function useRoundsEndingSoon() {
   });
 }
 
-// TODO: Filter + sort rounds (status, network, sort)
-export function useFilterRounds() {
-  return useRounds({});
-}
-
 //
```
Comment (vacekj) [inline packages/grant-explorer/src/features/api/rounds.ts:161]: Probably a leftover, or unfinished comment?

**Review by vacekj — APPROVED** [review]
Great work - left some comments, otherwise LGTM.

**Comment by melissa-neira** [comment]
@carlbarrdahl is there a way for the logic here to be a little different? here I've only selected a sorting option. I haven't selected anything to filter (it's in its default state) but yet the title changes to "multiple" which is contradictory as I've only selected a sorting option. as a user, I'd expect to see the title as "oldest", not "multiple". "multiple" works when I have both a sorting option and a filter selected. let me know if that makes sense!
Screenshot 2023-11-02 at 12 37 16

**Review by michellema1208 — COMMENTED** [review]
@carlbarrdahl amazing progress!! This is looking better and better every time I test it. 

Once quick thing I noticed is that when I select filters where "status=Active" and "status=Taking Applications", I'm seeing some rounds come up that have negative days left in the round. I would expect to only see rounds with >0 days left in the round show up here. 

Similarly, once those filters are working, would expect to have them selected by default if a user clicks "Explore Rounds" tab, so that we filter out past rounds by default. 

Screen Shot 2023-11-02 at 8 16 47 AM

**Comment by carlbarrdahl** [comment]
> @carlbarrdahl amazing progress!! This is looking better and better every time I test it.
> 
> Once quick thing I noticed is that when I select filters where "status=Active" and "status=Taking Applications", I'm seeing some rounds come up that have negative days left in the round. I would expect to only see rounds with >0 days left in the round show up here.
> 
> Similarly, once those filters are working, would expect to have them selected by default if a user clicks "Explore Rounds" tab, so that we filter out past rounds by default.
> 
> Screen Shot 2023-11-02 at 8 16 47 AM

Thank you @michellema1208!

I've fixed the filtering to not show past rounds.
I'm trying to figure out how to handle selecting multiple status filters. For example:
What happens is selecting all three: active, taking applications, and finished?

**Comment by melissa-neira** [comment]
> @carlbarrdahl is there a way for the logic here to be a little different? here I've only selected a sorting option. I haven't selected anything to filter (it's in its default state) but yet the title changes to "multiple" which is contradictory as I've only selected a sorting option. as a user, I'd expect to see the title as "oldest", not "multiple". "multiple" works when I have both a sorting option and a filter selected. let me know if that makes sense! Screenshot 2023-11-02 at 12 37 16

@carlbarrdahl did you get a chance to look at this comment?

**Comment by melissa-neira** [comment]
@carlbarrdahl hey, I noticed a little something here, outlined in this [loom](https://www.loom.com/share/bc58bc13e78042c3812f23709020c0d6?sid=4e400114-8c54-4d66-ba61-3068df0d759c)
everything else looks good to me!

**Review by melissa-neira — APPROVED** [review]
lgtm!!

**Pull request merged**